translator/trace/jaeger/traces_to_jaegerproto.go

```diff
@@ -115,42 +115,64 @@ func resourceToJaegerProtoProcess(resource pdata.Resource) *model.Process {
 	}
 
 	process := model.Process{}
+	attrsCount := attrs.Cap()
 	if serviceName, ok := attrs.Get(conventions.AttributeServiceName); ok {
 		process.ServiceName = serviceName.StringVal()
-		attrs.Delete(conventions.AttributeServiceName)
+		attrsCount--
 	}
-	process.Tags = attributesToJaegerProtoTags(attrs)
+	if attrsCount == 0 {
+		return &process
+	}
+
+	tags := make([]model.KeyValue, 0, attrsCount)
+	process.Tags = appendTagsFromResourceAttributes(tags, attrs)
 	return &process
 
 }
 
-func attributesToJaegerProtoTags(attrs pdata.AttributeMap) []model.KeyValue {
+func appendTagsFromResourceAttributes(dest []model.KeyValue, attrs pdata.AttributeMap) []model.KeyValue {
```

Comment:
Simpler code:
```
process.Tags = appendTagsFromResourceAttributes(attrsCount, attrs)
return &process

func appendTagsFromResourceAttributes(attrs pdata.AttributeMap) []model.KeyValue {
	if attrs.Cap() == 0 {
		return nil
	}
        dest = make([]model.KeyValue, 0, attrs.Cap())
}
```

Reply:
It's not always the case that we can use `attrs.Cap()` as capacity for tags slice, that's why I moved it outside

Reply:
Hmm, confused, why, because we allocate 1 extra element?

Reply:
The similar function `appendTagsFromAttributes` needs to allocate more than `attrs.Cap()`, I kept the same approach here as well